thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/FetchMetricDataInRangeAndOutputCSV.java

```diff
@@ -0,0 +1,109 @@
+  private static final String FILTERS = "filters";
+
+  private static final String DEFAULT_OUTPUT_FOLDER = "/home/ychung/Desktop/";
+  /**
+   * Fetch metric historical data from server and parse the json object
+   * @param args List of arguments
+   *             0: dataset/collection name
+   *             1: metric name
+   *             2: retrieval start time in ISO format, e.g. 2016-01-01T12:00:00
+   *             3: retrieval end time in ISO format
+   *             4: Aggregation time granularity, DAYS, HOURS,
+   *             5: dimensions drill down, ex: product,channel
+   *             6: filter in json format, ex: {"channel":["guest-email","guest-sms"]}
+   */
+  public static void main(String[] args){
+//    if(args.length < 5){
+//      System.out.println("Error: Insufficient number of arguments");
+//      return;
+//    }
+    String path2PersistenceFile = "/home/ychung/workspace/thirdeye-configs/local-configs/persistence.yml";
+    String dataset = "invite_sends_v2_additive";
+    String metric = "m2g_invite_sent";
+    String currentStartISO = "2016-01-01T12:00:00";
+    String currentEndISO = "2017-01-10T12:00:00";
+    String aggTimeGranularity = "DAYS";
+    String dimensions = "product";
+    String filters = "{\"channel\":[\"guest-email\"]}";
+    DateTimeFormatter fmt = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm");
+
+    if(FetchMetricDataAndExistingAnomaliesTool.TimeGranularity.fromString(aggTimeGranularity) == null){
+      System.out.println("Illegal time granularity");
+      return;
+    }
+
+
+    String filename = metric + "_" + currentStartISO + "_" + currentEndISO + ".csv";
+    Map<String, Map<DateTime, Integer>> metricContent;
+    try {
+      FetchMetricDataAndExistingAnomaliesTool thirdEyeDAO = new FetchMetricDataAndExistingAnomaliesTool(new File(path2PersistenceFile));
+      metricContent = thirdEyeDAO.fetchMetric(DEFAULT_HOST, Integer.valueOf(DEFAULT_PORT), dataset,
+          metric, currentStartISO, currentEndISO,
+          FetchMetricDataAndExistingAnomaliesTool.TimeGranularity.fromString(aggTimeGranularity), dimensions, filters);
+
+      String fname = metric + "_" + currentStartISO + "_" + currentEndISO + ".csv";
+      BufferedWriter bw = new BufferedWriter(new FileWriter(DEFAULT_OUTPUT_FOLDER + fname));
+
+      List<String> keys = new ArrayList<>(metricContent.keySet());
+      List<DateTime> dateTimes = new ArrayList<>(metricContent.get(keys.get(0)).keySet());
+      Collections.sort(dateTimes);
+
+      // Print Header
+      for(String str: keys){
+        bw.write("," + str);
+      }
+      bw.newLine();
+
+      for(DateTime dt : dateTimes){
+        bw.write(fmt.print(dt));
+        for(String key : keys){
+          Map<DateTime, Integer> map = metricContent.get(key);
+          bw.write("," + map.get(dt));
+        }
+        bw.newLine();
+      }
+      bw.close();
+    }
+    catch (Exception e){
+      System.out.println(e.getMessage());
```

Comment:
Use Logger instead of sysout everywhere. and then LOG.error("msg", e) or LOG.info("message {}", arg1);